internal/statestore/public.go

```diff
@@ -84,6 +84,15 @@ type Service interface {
 
 	// NewMutex returns an interface of a new distributed mutex with given name
 	NewMutex(key string) RedisLocker
+
+	// AcknowledgeBackfill - store Backfill's last accessed time
+	AcknowledgeBackfill(ctx context.Context, id string) error
```

Comment:
nit: I suggest sticking to the current comments format like:
_AcknowledgeBackfill stores Backfill's last accessed time_

Reply:
// AcknowledgeBackfill - stores Backfill's last acknowledgement time